core/remotes/resolver.go

```diff
@@ -109,3 +113,77 @@ func WithMediaType(mediatype string) FetchByDigestOpts {
 		return nil
 	}
 }
+
+////
+// Layer fetch config
+////
+
+type FetcherConfig struct {
+	MaxConcurrentDownloads         int
+	MaxConcurrentDownloadsPerLayer int
+	ConcurrentDownloadChunkSize    int
+	Limiter                        limiter
```

Comment:
why does the `FetcherConfig` need to have both `int` limit configs as well as an actual `limiter`?

Reply:
The limiter is here so that two downloads running at the same time use the same limiter.

The rest is the config, ex, to tell for example how many goroutines could be started at once.

Reply:
I kinda wanted to wrap the limiter with something that could give it's own limit, to avoid that, and to take the number from it. Didn't do it to not add too much code.